core/src/main/java/org/jdbi/v3/core/mapper/EnumMapper.java

```diff
@@ -78,6 +81,11 @@ public E map(ResultSet rs, int columnNumber, StatementContext ctx) throws SQLExc
         private static Object getValueByName(Class<? extends Enum<?>> enumClass, String name) {
             final Enum<?>[] enumConstants = enumClass.getEnumConstants();
             return JdbiOptionals.findFirstPresent(
+                    () -> Arrays.stream(enumConstants).filter(e -> {
```

Comment:
If somebody annotates an enum with `@DatabaseValue` then I don't think we should be falling back to matching the enum name.

Reply:
https://github.com/jdbi/jdbi/pull/1566

Reply:
> If somebody annotates an enum with `@DatabaseValue` then I don't think we should be falling back to matching the enum name.

If I understand you correctly, given the already included enum
```
private enum Foobar {
    BAR, FOO, @DatabaseValue("CUST") CUSTOM
}
```
this test should pass (currently it doesn’t) and be included:
```
@Test
public void customizedNamesAreNotMappedToEnumName() throws NoSuchFieldException {
    db.getJdbi().useHandle(h -> {
        Optional<Foobar> mappedEnum = h.createQuery("select :name")
                .bind("name", "CUSTOM")
                .mapTo(Foobar.class)
                .findOne();

        assertThat(mappedEnum)
                .isEmpty();
    });
}
```